functions.php

```diff
@@ -53,6 +53,22 @@ function as_schedule_single_action( $timestamp, $hook, $args = array(), $group =
  * @return int The action ID.
  */
 function as_schedule_recurring_action( $timestamp, $interval_in_seconds, $hook, $args = array(), $group = '', $unique = false ) {
+	$interval = (int) $interval_in_seconds;
+
+	// We expect an integer and allow it to be passed using float and string types, but otherwise
+	// should reject unexpected values.
+	if ( ! is_numeric( $interval_in_seconds ) || $interval_in_seconds != $interval ) {
+		_doing_it_wrong(
+			__METHOD__,
+			sprintf( 
+				/* translators: 1: provided value 2: provided type. */
+				__( 'An integer was expected but "%1$s" (%2%s) was received.', 'action-scheduler' ),
```

Comment:
Need to fix the second placeholder. Also, we can add a measure of safety by escaping the string:

```suggestion
				esc_html__( 'An integer was expected but "%1$s" (%2$s) was received.', 'action-scheduler' ),
```

Apologies for missing those things in the initial suggestion.